### ISSUE TITLE
Rebuild plugins when node version changes

### ISSUE THREAD

**Issue by olizilla** [ssbc/scuttlebot#453]
I've upgraded node a few times since I last ran sbot locally, as I've been enjoying ssb from the comfy seats of patchwork-classsic and patchwork which both start up an embedded sbot.

I now can't run `sbot server` as ssb-links needs a rebuild.

```sh
$ sbot server
/Users/oli/.ssb/node_modules/ssb-links/node_modules/bindings/bindings.js:83
        throw e
        ^

Error: The module '/Users/oli/.ssb/node_modules/ssb-links/node_modules/leveldown/build/Release/leveldown.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 48. This version of Node.js requires
NODE_MODULE_VERSION 57. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (module.js:653:18)
```

I can't uninstall ssb-links from the cli either

```sh
$ sbot plugins.uninstall ssb-links
Error: Could not connect to the scuttlebot server.
Use the "server" command to start it.
```

So I could nave/nvm the problem away or go and manually run run a rebuild in the `/Users/oli/.ssb/node_modules/ssb-links` dir, or maybe rm -rf the `.ssb/node_modules` dir but my first instinct is hesitation. I'm not used to seeing a node_modules dir there, so I'm less confident about messing with it, and it feels like sbot would benefit from a way of dealing with this, or at least, not getting caught up in this _fails either way_ scenario. 

Happy to help, but could use some directions.

**Comment by olizilla** [comment]
`rebuild` and `rm -rf` both caused me more errors. Obvs, I should have just disabled it in the `~.ssb/config` file.

**Comment by twoangrybears** [comment]
What happens when you try to run sbot now?

I had a similar situation and was able to install the plugins sbot was complaining about as follows -

    cd ~/.ssb  
    npm i ssb-fulltext 
    npm i ssb-about 
    npm i ssb-backlinks

This let me start sbot and enable them -

    sbot plugins.enable ssb-fulltext
    sbot plugins.enable ssb-about
    sbot plugins.enable ssb-backlinks

Then I had to stop and restart sbot.

Does this help? 

If not perhaps you could list what you describe as "more errors" and someone who knows more than me might be able to help.

**Comment by dominictarr** [comment]
first, backup your ~/.ssb directory, then `rm -rf ~/.ssb/flume/*.json ~/.ssb/flume/*/*` then restart patchwork@3, I think that will probably do it?

hmm, actually if you are running patchwork@3 it shouldn't need ~/.ssb/node_modules at all, maybe try just deleting those and clearing the plugins section of the config?

**Comment by olizilla** [comment]
Thanks, I was able to fix it by disabling the plugins in `~/.ssb/config`. That allowed me to start ssb. What I really wanted was to surface a UX issue, though I think it's really a node UX problem. 

The frustration I hit was trying to fix the problem from within the `sbot` command, but a plugin that needed a rebuild meant i couldn't run `sbot`; even to remove the plugin. I should have checked the config file sooner as I obviously added them in there at some point, but it was ages ago, and I did it by following some steps that someone posted with no understanding. Difficult to fix that tendency in people tho...

sbot currently has no control over which node version it's running on, so this problem will crop up again. I figure the only issue here is documenting what the purpose of the files in the `.ssb` folder are, perhaps catching errors during plugin init, if possible, so we can give a helpful error message to explain how to disable the plugin causing the issue... It'd be rad to add to post error messages with a link to an ssb post, like decentralised [url addressable errors](https://rauchg.com/2016/addressable-errors)... though that would be tricky as we'd want current users sbot to post a message, and it's the the thing that is erroring. Anyhoo, you get the idea.
